webapp/src/components/Metrics/PerformanceAnalysisTable.tsx

```diff
@@ -0,0 +1,608 @@
+              comparedPipeline: comparedPipelineData.metricsOverall[0],
+            }),
+        },
+        ...basePipelineData.metricsPerFilter[selectedMetricPerFilterOption].map(
+          (basePipeline, index) => ({
+            id: index,
+            basePipeline,
+            ...(comparedPipeline &&
+              comparedPipelineData && {
+                comparedPipeline:
+                  comparedPipelineData.metricsPerFilter[
+                    selectedMetricPerFilterOption
+                  ][index],
+              }),
+          })
+        ),
+      ]) ||
+      []
+    );
+  }, [
+    basePipelineData,
+    comparedPipeline,
+    comparedPipelineData,
+    selectedMetricPerFilterOption,
+  ]);
+
+  const columns: Column<Row>[] = React.useMemo(() => {
+    const metricsEntries = Object.entries(metricInfo ?? {});
+
+    const customSort = (
+      // Use this sort to keep the overall row at the top always.
+      // All columns must use it as their sorter.
+      v1: GridCellValue,
+      v2: GridCellValue,
+      param1: GridSortCellParams,
+      param2: GridSortCellParams
+    ) => {
+      const sign = sortModel[0].sort === "desc" ? 1 : -1;
+      //Custom sort to keep 'overall' at top
+      if ((param1.id as number) === OVERALL_ROW_ID) return sign;
+      if ((param2.id as number) === OVERALL_ROW_ID) return -sign;
+
+      // Fall back to default sort
+      return gridStringOrNumberComparator(v1, v2, param1, param2);
+    };
+
+    const METRIC_COLUMN = {
+      flex: 1,
+      minWidth: 150,
+      maxWidth: 250,
+      sortComparator: customSort,
+    };
+
+    const groupHeader = (
+      baseOrCompared: typeof pipelines[number],
+      shortHeader: React.ReactNode,
+      longHeader: React.ReactNode = shortHeader
+    ) => ({
+      ...(comparedPipeline === undefined
+        ? { renderHeader: () => shortHeader }
+        : {
+            headerClassName: "span",
+            cellClassName: "span",
+            renderHeader:
+              baseOrCompared === "basePipeline"
+                ? ({ colDef }: GridColumnHeaderParams) => (
+                    <>
+                      {config?.pipelines?.[pipeline.pipelineIndex].name}
+                      <Box
+                        position="absolute"
+                        left={0}
+                        right={-2 * colDef.computedWidth}
+                        top={0}
+                        lineHeight={1}
+                        textAlign="center"
+                      >
+                        {longHeader}
+                      </Box>
+                    </>
+                  )
+                : () => config?.pipelines?.[comparedPipeline].name,
+          }),
+    });
+
+    const utteranceCount: Column<Row>[] =
+      selectedMetricPerFilterOption === "label"
+        ? [
+            {
+              field: "utteranceCount",
+              headerName: "Total",
+              description: "Number of Utterances in each model",
```

Comment:
There aren't really utterances "in" a model. Also, we prefer using "pipeline". That being said, none of the other columns have that kind of mention. It seems straightforward. I think we can remove it.
```suggestion
              description: "Number of utterances",
```

Reply:
True!. The design suggested displaying it when the "Label" column is chosen. It was recommended to display in the header, but considering the width of the header, I ended up adding it as a tooltip.